packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -51,8 +57,42 @@ const defaultProps = {
 
 // eslint-disable-next-line react/prefer-stateless-function
 class DatePickerInput extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleOnButtonClick = this.handleOnButtonClick.bind(this);
+    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+  }
+
+  handleOnButtonClick() {
+    // If the containing component (e.g. modal) has the focus trapped, the focus needs to be released so that the picker can get focus.
+    if (this.props.app && this.props.app.releaseFocus) {
+      this.props.app.releaseFocus();
```

Comment:
@dkasper-was-taken requestFocus now means that popup component is requesting focus and the modal should give up the focus trap. releaseFocus means the popup component no longer needs focus and the modal can regain the focus trap.

https://github.com/cerner/terra-core/pull/640/commits/8eddd676bf0c82b1ecd3750df7ff8e288104aab3